cli/cmd/sources.go

```diff
@@ -330,6 +374,128 @@ var sourceStatusCmd = &cobra.Command{
+	msg := "enable"
+	if disableInstrumentation {
+		msg = "disable"
+	}
+
+	return &cobra.Command{
+		Use:     fmt.Sprintf("%s [name]", workload.WorkloadKindLowerCaseFromKind(workloadKind)),
+		Short:   fmt.Sprintf("%s a %s for Odigos instrumentation", msg, workloadKind),
+		Long:    fmt.Sprintf("This command %ss the provided %s for Odigos instrumentatin. It will create a Source object if one does not already exists, or update the existing one if it does.", msg, workloadKind),
+		Args:    cobra.ExactArgs(1),
+		Aliases: kindAliases[workloadKind],
+		Run: func(cmd *cobra.Command, args []string) {
+			enableOrDisableSource(cmd, args, workloadKind, disableInstrumentation)
+		},
+	}
+}
+
+func updateOrCreateSourceForObject(ctx context.Context, client *kube.Client, workloadKind k8sconsts.WorkloadKind, argName string, disableInstrumentation bool) (*v1alpha1.Source, error) {
+	var err error
+	obj := workload.ClientObjectFromWorkloadKind(workloadKind)
+	var objName, objNamespace, sourceNamespace string
+	switch workloadKind {
+	case k8sconsts.WorkloadKindDaemonSet:
+		obj, err = client.Clientset.AppsV1().DaemonSets(sourceNamespaceFlag).Get(ctx, argName, metav1.GetOptions{})
+		objName = obj.GetName()
+		objNamespace = obj.GetNamespace()
+		sourceNamespace = sourceNamespaceFlag
+	case k8sconsts.WorkloadKindDeployment:
+		obj, err = client.Clientset.AppsV1().Deployments(sourceNamespaceFlag).Get(ctx, argName, metav1.GetOptions{})
+		objName = obj.GetName()
+		objNamespace = obj.GetNamespace()
+		sourceNamespace = sourceNamespaceFlag
+	case k8sconsts.WorkloadKindStatefulSet:
+		obj, err = client.Clientset.AppsV1().StatefulSets(sourceNamespaceFlag).Get(ctx, argName, metav1.GetOptions{})
+		objName = obj.GetName()
+		objNamespace = obj.GetNamespace()
+		sourceNamespace = sourceNamespaceFlag
+	case k8sconsts.WorkloadKindNamespace:
+		obj, err = client.Clientset.CoreV1().Namespaces().Get(ctx, argName, metav1.GetOptions{})
+		objName = obj.GetName()
+		objNamespace = obj.GetName()
+		sourceNamespace = obj.GetName()
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	var source *v1alpha1.Source
+	selector := labels.SelectorFromSet(labels.Set{
+		k8sconsts.WorkloadNameLabel:      obj.GetName(),
+		k8sconsts.WorkloadNamespaceLabel: sourceNamespace,
+		k8sconsts.WorkloadKindLabel:      string(workloadKind),
+	})
+	sources, err := client.OdigosClient.Sources(sourceNamespace).List(ctx, metav1.ListOptions{LabelSelector: selector.String()})
+	if len(sources.Items) > 0 {
+		source = &sources.Items[0]
+	} else {
+		source = &v1alpha1.Source{
+			ObjectMeta: v1.ObjectMeta{
+				GenerateName: workload.CalculateWorkloadRuntimeObjectName(objName, workloadKind),
+				Namespace:    sourceNamespace,
+			},
+			Spec: v1alpha1.SourceSpec{
+				Workload: k8sconsts.PodWorkload{
+					Kind:      workloadKind,
+					Name:      objName,
+					Namespace: objNamespace,
+				},
+			},
+		}
+
+	}
+
+	source.Spec.DisableInstrumentation = disableInstrumentation
```

Comment:
I wonder if it's easy and useful to denote a "no-op" operation (e.g. the source was effectively uninstrumented and this command keeps it as is, or source was intrumented and is now uninstrumented).

so if user runs a command that effectively does not change the instrumentation state, it might be a mistake and the user should not expect any change in odigos.

Not sure it's easy to achieve for all the possible states, but if it does, can be nice-to-have it :)